torch/_inductor/triton_heuristics.py

```diff
@@ -536,8 +536,9 @@ def _find_names(obj):
     import inspect
 
     frame = inspect.currentframe()
-    for frame in iter(lambda: frame.f_back, None):  # type: ignore[union-attr]
+    while frame is not None:
         frame.f_locals
+        frame = frame.f_back
```

Comment:
This code is a bit magical for me. I suspect traversing through frames and calling `frame.f_locals` creates new objects that are accessible via `gc.get_referrers`, and my change doesn't break anything. I don't have CUDA to confirm that.

Reply:
This looks fine to me, but if anything breaks in this PR, it's likely this code.